npe2/manifest/_npe1_shim.py

```diff
@@ -0,0 +1,113 @@
+import logging
+import os
+import site
+import warnings
+from pathlib import Path
+from shutil import rmtree
+from typing import List, Sequence
+
+from appdirs import user_cache_dir
+
+from .._from_npe1 import manifest_from_npe1
+from .package_metadata import PackageMetadata
+from .schema import PluginManifest, discovery_blocked
+
+try:
+    from importlib import metadata
+except ImportError:
+    import importlib_metadata as metadata  # type: ignore
+
+
+logger = logging.getLogger(__name__)
+SHIM_CACHE = Path(user_cache_dir("napari", "napari")) / "npe2" / "shims"
+NPE2_NOCACHE = "NPE2_NOCACHE"
+
+
+def clear_cache(names: Sequence[str] = ()) -> List[Path]:
+    """Clear cached npe1 shim files
+
+    Parameters
+    ----------
+    names : Sequence[str], optional
+        selection of plugin names to clear, by default, all will be cleared
+
+    Returns
+    -------
+    List[Path]
+        List of filepaths cleared
+    """
+    _cleared: List[Path] = []
+    if SHIM_CACHE.exists():
+        if names:
+            for f in SHIM_CACHE.glob("*.yaml"):
+                if any(f.name.startswith(f"{n}_") for n in names):
+                    f.unlink()
+                    _cleared.append(f)
+        else:
+            _cleared = list(SHIM_CACHE.iterdir())
+            rmtree(SHIM_CACHE)
+    return _cleared
+
+
+class NPE1Shim(PluginManifest):
+    _is_loaded: bool = False
+    _dist: metadata.Distribution
+
+    def __init__(self, dist: metadata.Distribution):
+        meta = PackageMetadata.from_dist_metadata(dist.metadata)
+        super().__init__(name=dist.metadata["Name"], package_metadata=meta)
+        self._dist = dist
+
+    def __getattribute__(self, __name: str):
+        if __name == "contributions" and not self._is_loaded:
+            self._load_contributions()
+        return super().__getattribute__(__name)
+
+    def _load_contributions(self) -> None:
+        """imports and inspects package using npe1 plugin manager"""
+
+        self._is_loaded = True  # if we fail once, we still don't try again.
+        if self._cache_path().exists() and not os.getenv(NPE2_NOCACHE):
+            mf = PluginManifest.from_file(self._cache_path())
+            self.contributions = mf.contributions
+            logger.debug("%r npe1 shim loaded from cache", self.name)
```

Comment:
First, I love the use of logger.debug.  Yay for logging.

Second, I think it's fine for internal use but I'm a bit worried that the term `npe1` is something devs will understand but users/new-folks won't? not sure if that matters for a debug message.

Reply:
left for debug... but changed everywhere else to "first generation napari plugin"